pkg/sql/exec/cast_tmpl.go

```diff
@@ -134,6 +134,7 @@ func (c *castOp_FROMTYPE_TOTYPE) Next(ctx context.Context) coldata.Batch {
 				}
 			}
 		} else {
+			col = col[:n]
```

Comment:
This needs to be `col = execgen.SLICE(col, 0, int(n))`, since not all types will be normally sliceable like this.